### PR TITLE
BitcoinWallet implementation

### PR DESCRIPTION
**Pull request by szerepak**

### CODE REVIEW

File: clove/utils/network.py [hunk 1]
```diff
@@ -7,3 +7,19 @@ def hostname_resolves(hostname):
         return True
     except socket.error:
         return False
+
+
+def left_pad(s, length=64):
```
Comment (trojkat) [inline clove/utils/network.py]: Those methods are not related to networking, we should move them to a separate file.

File: clove/network/bitcoin.py [hunk 2]
```diff
@@ -166,3 +172,42 @@ class TestNetBitcoin(Bitcoin):
         'testnet-seed.bluematt.me',
     )
     port = 18333
+
+
+class BitcoinWallet(object):
+
+    private_key = None
+    public_key = None
+    password = b''
```
Comment (trojkat) [inline clove/network/bitcoin.py]: Those fields are unused, right?

File: clove/network/bitcoin.py [hunk 3]
```diff
@@ -166,3 +172,42 @@ class TestNetBitcoin(Bitcoin):
...
+
+    def __init__(self):
```
Comment (trojkat) [inline clove/network/bitcoin.py]: I think that `__init__` should be able to support two other scenarios: 
1. take `private_key`
```
>> wallet = BitcoinWallet(private_key='cSYq9JswNm79GUdyz6TiNKajRTiJEKgv4RxSWGthP3SmUHiX9WKe')
>> wallet.get_address()
>> 'msJ2ucZ2NDhpVzsiNE5mGUFzqFDggjBVTM'
```
2. take `encrypted_private_key` and `password`
```
>> wallet = BitcoinWallet(encrypted_private_key='4327432432hui432h432432v432', password='foo')
>> wallet.get_address()
>> 'msJ2ucZ2NDhpVzsiNE5mGUFzqFDggjBVTM'
```

File: tests/test_wallet.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    BitcoinWallet(private_key='KxhniiXPCdBBpJmQnYPHmutKJq42Wm3yPY6AAKxvDPnTt8KA8BJF'),
+    BitcoinWallet(
+        encrypted_private_key=b'3bktZ1EG4dvKOilVbveJo8WoScrVJqGOfhjULdfooL'
+                              b'CN7Il5Bu4CCA0HBP1k7iPZWyfohxiSdwJ3CpgHijIL2zb1THA=',
+        password='test_password_xyz'
+    )
+])
+def test_bitcoin_wallet(wallet):
+    wallet = BitcoinWallet()
+    assert isinstance(wallet.private_key, CBitcoinSecret) is True
```
Comment (trojkat) [inline tests/test_wallet.py]: `assert something_true` is enough.

File: clove/network/bitcoin.py [hunk 5]
```diff
@@ -166,3 +172,51 @@ class TestNetBitcoin(Bitcoin):
...
+    def __init__(self, private_key=None, encrypted_private_key=None, password=None):
+        if private_key is None and encrypted_private_key is None:
+            secret = ''.join(choices(ascii_letters + digits, k=64))
+            self.secret = sha256(bytes(secret.encode('utf-8'))).digest()
+            self.private_key = CBitcoinSecret.from_secret_bytes(secret=self.secret)
+
+        elif private_key is not None:
+            self.private_key = CBitcoinSecret(private_key)
+
+        elif encrypted_private_key is not None and password is not None:
```
Comment (trojkat) [inline clove/network/bitcoin.py:188]: or just
`elif not encrypted_private_key and not password:`

**Review by trojkat — APPROVED** [review]

**Review by pafff — APPROVED** [review]
